File: Interviews/GFG/Daily-Problems/2022/October/Day-14/Maximize-the-sum-of-selected-numbers-from-an-array-to-make-it-empty.cpp
class Solution {
public:
	int maximizeSum(int a[], int n) {
		vector<int> mark(100001, 0);
		for (int i = 0; i < n; i++) {
			mark[a[i]] += 1;
		}

		int ans = 0;
		for (int i = 100000; i >= 1; i--) {
			ans += mark[i] * i;
			mark[i - 1] = max(mark[i - 1] - mark[i], 0);
		}

		return ans;
	}
};

File: Interviews/GFG/Daily-Problems/2022/October/Day-14/Maximize-the-sum-of-selected-numbers-from-an-array-to-make-it-empty.py
class Solution:
    def maximizeSum (self, arr, n) : 
        mark, ans, maxx = [0 for i in range(100001)], 0, 0
        for i in arr:
            mark[i] += 1
            maxx = max(maxx, i)
        
        for i in range(maxx, 0, -1):
            ans += mark[i] * i
            mark[i - 1] = max(mark[i - 1] - mark[i], 0)
        
        return ans